datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -48,22 +58,35 @@ private DurationDeserializer()
     @Override
     public Duration deserialize(JsonParser parser, DeserializationContext context) throws IOException
     {
+        String string = parser.getText().trim();
         switch (parser.getCurrentTokenId())
         {
             case JsonTokenId.ID_NUMBER_FLOAT:
                 BigDecimal value = parser.getDecimalValue();
+                // If the decimal isnt within the bounds of a float, bail out
+                if(value.compareTo(INSTANT_MAX) > 0 ||
+                        value.compareTo(INSTANT_MIN) < 0) {
+                    NumberFormat formatter = new DecimalFormat("0.0E0");
+                    throw new JsonParseException(context.getParser(),
+                            String.format("Value of BigDecimal (%s) not within range to be converted to Duration",
+                                    formatter.format(value)));
```

Comment:
Could this use `string` rather than reformatting the parsed value?

Reply:
If the string is a 1,000,000 characters long, I would assume printing it might be more trouble than what its worth, --  eg the test case where I append "1000000000" a thousand times would end up with a 1mb long exception message.

I can just not print the string and just say its not within the expected range.

Reply:
I think it'd be better to just include length of String as presumably this should only be hit for attack, or some other use case in which input is not legitimate. So printing that value does not make sense.

Reply:
Makes sense.

Reply:
The DoS is not coupled to the length of the string, in fact the problem is that short input strings can cause the failure, by representing very large numbers.